Simplify domain membership logic

esoergel opened the pull request:
I think this was a holdover from multi-domain organizations.  It did some weird permissions aggregations stuff that was getting in my way.  This should have no functional differences, but it seems like an important enough change that it deserved its own PR.

esoergel committed:
Simplify domain membership logic

I think this was a holdover from multi-domain organizations.  It did
some weird permissions aggregations stuff that was getting in my way.

esoergel commented on corehq/apps/users/models.py:
```diff
@@ -2029,30 +2020,11 @@ def get_role(self, domain=None, include_teams=True, checking_global_admin=True):
-        if domain_memberships:
-            for domain_membership, membership_source in domain_memberships:
-                permission = domain_membership.permissions
-                total_permission |= permission
```
This bit is the important part.  I assume at some point some users must've had multiple domain memberships, but now this function is only called in two places, both of which pass in a list of length 1 for `domain_memberships` (and both are modified in this PR).

What remained was this weird behavior, where we'd take their assigned permissions and compare that against the default `Permissions()` instance, and creating a new, fake user role with the more permissive of each property.  This works fine now, since the default permissions all disallow everything, but it's very strange that users don't actually get the role they're assigned, exactly.

Additionally, if we ever add any permissions that default to the *more* permissive option (as is the case in my unrelated change), this would always give the users that permission, even when their role expressly disallows it.

czue commented:
nice find. this code had a bit of a... stank ... to it?

build fail looks real.
```
======================================================================
ERROR: test_create (corehq.apps.api.tests.TestWebUserResource)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/mnt/commcare-hq-ro/corehq/apps/api/tests.py", line 723, in test_create
    content_type='application/json')
  File "/mnt/commcare-hq-ro/corehq/apps/api/tests.py", line 200, in _assert_auth_post_resource
    response = self.client.post(api_url, post_data, content_type=content_type)
  File "/vendor/lib/python2.7/site-packages/django/test/client.py", line 482, in post
    secure=secure, **extra)
  File "/vendor/lib/python2.7/site-packages/django/test/client.py", line 296, in post
    secure=secure, **extra)
  File "/vendor/lib/python2.7/site-packages/django/test/client.py", line 358, in generic
    return self.request(**r)
  File "/vendor/lib/python2.7/site-packages/django/test/client.py", line 440, in request
    six.reraise(*exc_info)
  File "/vendor/lib/python2.7/site-packages/django/core/handlers/base.py", line 111, in get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/vendor/lib/python2.7/site-packages/django/views/decorators/csrf.py", line 57, in wrapped_view
    return view_func(*args, **kwargs)
  File "/vendor/lib/python2.7/site-packages/tastypie/resources.py", line 219, in wrapper
    response = callback(request, *args, **kwargs)
  File "/vendor/lib/python2.7/site-packages/tastypie/resources.py", line 450, in dispatch_list
    return self.dispatch('list', request, **kwargs)
  File "/mnt/commcare-hq-ro/corehq/apps/api/resources/__init__.py", line 99, in dispatch
    return super(HqBaseResource, self).dispatch(request_type, request, **kwargs)
  File "/vendor/lib/python2.7/site-packages/tastypie/resources.py", line 482, in dispatch
    response = method(request, **kwargs)
  File "/vendor/lib/python2.7/site-packages/tastypie/resources.py", line 1390, in post_list
    updated_bundle = self.full_dehydrate(updated_bundle)
  File "/vendor/lib/python2.7/site-packages/tastypie/resources.py", line 895, in full_dehydrate
    data[field_name] = method(bundle)
  File "/mnt/commcare-hq-ro/corehq/apps/api/resources/v0_1.py", line 285, in dehydrate_role
    return bundle.obj.get_role(bundle.request.domain).name
AttributeError: 'NoneType' object has no attribute 'name'
```

esoergel committed:
Don't fail when no role is assigned

esoergel committed:
has_permission and get_role are useless

Now that they're simplified, you can see that they behave basically the
same as their super methods.

esoergel commented:
@millerdev @NoahCarnahan
